hooks/useSound.tsx

```diff
@@ -0,0 +1,254 @@
+      } else {
+        throw new Error("Unsupported chainId");
+      }
+
+      const _client = SoundClient({
+        signer: mySigner,
+        soundCreatorAddress: contractAddresses.mainnet.soundCreatorV1,
+      });
+
+      setClient(_client);
+    }
+
+    initSoundClient();
+  }, [connector, signer]);
+
+  const generateProposalPayload = async () => {
+    if (!client) return;
+    if (!address) return;
+    if (!signer) return;
+    if (!soundCreatorAddress) return;
+
+    dispatch({ type: "GENERATE_BYTECODE" });
+
+    const customSalt = "0x" + Math.random().toString(16).slice(2);
+
+    const editionConfig: EditionConfig = {
+      name: state.songName,
+      symbol: state.songSymbol,
+      metadataModule: state.metadataModule,
+      contractURI: state.contractUri,
+      baseURI: state.baseUri,
+      fundingRecipient: state.fundingRecipient,
+      royaltyBPS: state.royaltyBps,
+      editionMaxMintableUpper: state.editionMaxMintable,
+      editionMaxMintableLower: 0,
+      editionCutoffTime: state.editionCutoffTime,
+      shouldFreezeMetadata: false,
+      shouldEnableMintRandomness: true,
+      enableOperatorFiltering: true,
+    };
+
+    const mintConfigs: MintConfig[] = [];
+
+    switch (MinterType[state.minter]) {
+      case "RANGE_MINTER":
+        mintConfigs.push({
+          mintType: "RangeEdition",
+          cutoffTime: state.editionCutoffTime,
+          minterAddress: contractAddresses.mainnet.rangeEditionMinter,
+          price: 0,
+          startTime: 0,
+          endTime: 0,
+          affiliateFeeBPS: 0,
+          maxMintableUpper: state.editionMaxMintable,
+          maxMintableLower: 0,
+          maxMintablePerAccount: 1,
+        });
+        break;
+      default:
+        break;
+    }
+
+    // Transaction
+    const txnOverrides: Overrides = {
+      gasLimit,
+      maxFeePerGas,
+      maxPriorityFeePerGas,
+    };
+
+    const formattedSalt = getSaltAsBytes32(
+      customSalt || Math.random() * 1_000_000_000_000_000,
+    );
+
+    // Precompute the edition address.
+    const [editionAddress, _] = await SoundCreatorV1__factory.connect(
+      soundCreatorAddress,
+      signer,
+    ).soundEditionAddress(address, formattedSalt);
+
+    console.log("Precompute Edition Address ", editionAddress);
+
+    const editionInterface = ISoundEditionV1_1__factory.createInterface();
+
+    /**
+     * Encode all the bundled contract calls.
+     */
+    const contractCalls: ContractCall[] = [];
+
+    console.log("Grant Roles");
+    // Grant MINTER_ROLE for each minter.
+    const mintersToGrantRole = Array.from(
+      new Set(mintConfigs.map((m) => m.minterAddress)),
+    );
+    for (const minterAddress of mintersToGrantRole) {
+      contractCalls.push({
+        contractAddress: editionAddress,
+        // TODO: The new contract interface removed grantRoles, but there should be an equivalent function.
+        calldata: editionInterface.encodeFunctionData("grantRoles", [
```

Comment:
needs fix to make it build. I'm not sure if they require granting roles anymore in the new contracts